orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -179,6 +189,135 @@ class TaskController {
+  ) {
+    if (triggerTimeStartBoundary < 0) {
+      throw new RuntimeException(String.format("triggerTimeStartBoundary must be >= 0: triggerTimeStartBoundary=%s", triggerTimeStartBoundary))
+    }
+    if (triggerTimeEndBoundary < 0) {
+      throw new RuntimeException(String.format("triggerTimeEndBoundary must be >= 0: triggerTimeEndBoundary=%s", triggerTimeEndBoundary))
+    }
+    if (triggerTimeStartBoundary > triggerTimeEndBoundary) {
+      throw new RuntimeException(String.format("triggerTimeStartBoundary must be <= triggerTimeEndBoundary: triggerTimeStartBoundary=%s, triggerTimeEndBoundary=%s", buildTimeStartBoundary, buildTimeEndBoundary))
+    }
+    if (startIndex < 0) {
+      throw new RuntimeException(String.format("startIndex must be >= 0: startIndex=%s", startIndex))
+    }
+    if (size <= 0) {
+      throw new RuntimeException(String.format("size must be > 0: size=%s", size))
+    }
+
+    Map triggerParams
+    if (encodedTriggerParams != null) {
+      byte[] decoded = Base64.getDecoder().decode(encodedTriggerParams)
+      String str = new String(decoded, Charset.forName("UTF-8"))
+      triggerParams = mapper.readValue(str, Map.class)
+    } else {
+      triggerParams = new HashMap()
+    }
+
+    if (triggerType != "*") {
+      triggerParams.put("type", triggerType)
+    }
+    if (eventId != null) {
+      triggerParams.put("eventId", eventId)
+    }
+
+    statuses = statuses ?: ExecutionStatus.values()*.toString().join(",")
+
+    ExecutionRepository.BuildTimeBoundaryExecutionCriteria executionCriteria = new ExecutionRepository.BuildTimeBoundaryExecutionCriteria(
+      buildTimeStartBoundary: triggerTimeStartBoundary,
+      buildTimeEndBoundary: triggerTimeEndBoundary,
+      statuses: (statuses.split(",") as Collection)
+    )
+
+    List<String> pipelineConfigIds = application == "*" ? getPipelineConfigIdsOfReadableApplications() : front50Service.getPipelines(application, false)*.id as List<String>
+
+    List<Execution> pipelineExecutions = rx.Observable.merge(pipelineConfigIds.collect {
+      executionRepository.retrievePipelinesForPipelineConfigIdWithBuildTimeBoundary(it, executionCriteria).filter {
+        if (!StringUtils.isBlank(pipelineName) && pipelineName != it.name) {
+          return false
+        }
+
+        // Compare each execution's trigger to input triggerParams
+        Map triggerAsMap = mapper.convertValue(it.getTrigger(), Map.class)
+        return recursivelyCheckIfObjectFieldsMatchesAllSubsetFields(triggerAsMap, triggerParams)
+      }
+    }).subscribeOn(Schedulers.io())
+      .toList()
+      .toBlocking()
+      .single()
+      .sort(reverseBuildTime)
+
+    if (reverse) {
+      pipelineExecutions.reverse(true)
+    }
+
+    List<Execution> rval
+    if (startIndex >= pipelineExecutions.size()) {
+      rval = []
+    } else {
+      rval = pipelineExecutions.subList(startIndex, Math.min(pipelineExecutions.size(), startIndex + size))
+    }
+
+    if (!expand) {
```

Comment:
Often this kind of parameter exists to make the endpoint more performant -- could you add a TODO to perhaps handle this before making the expensive calls to filter & retrieve all executions?